File: 0459-repeated-substring-pattern/0459-repeated-substring-pattern.java
class Solution {
    public boolean repeatedSubstringPattern(String s) {
        int n = s.length();

        for (int len = 1; len <= n / 2; len++) {

            // Substring length must divide
            // the complete string length
            if (n % len != 0) {
                continue;
            }

            boolean valid = true;

            // Compare with the first substring
            for (int i = len; i < n; i++) {
                if (s.charAt(i) != s.charAt(i % len)) {
                    valid = false;
                    break;
                }
            }

            if (valid) {
                return true;
            }
        }

        return false;
    }
}

File: 0459-repeated-substring-pattern/0459-repeated-substring-pattern.c
#include <string.h>

bool repeatedSubstringPattern(char* s) {
    int n = strlen(s);

    // Try every possible substring length
    for (int len = 1; len <= n / 2; len++) {

        // n must be divisible by len
        if (n % len != 0)
            continue;

        bool match = true;

        // Compare the whole string with repeated substring
        for (int i = 0; i < n; i++) {
            if (s[i] != s[i % len]) {
                match = false;
                break;
            }
        }

        if (match)
            return true;
    }

    return false;
}